### PR TITLE
Feature watch task

### PR DESCRIPTION
**Pull request by kaiwollhaupt**
Merge branch feature watch-task into 2.x-dev .

### CODE REVIEW

File: grunt/postcss.js [hunk 1]
```diff
@@ -1,18 +1,39 @@
...
+        cssProduction: {
+            options: {
+                map: false,
                 processors: [
                     require( 'autoprefixer' )( {
                         browsers: [ 'last 2 versions', 'ie >= 8', 'Firefox >= 12', 'ios >= 7', 'android >= 4' ]
-                    } )
+                    } ),
+                    require( 'postcss-inline-svg' ),
+                    require( 'postcss-svgo' )
```
Comment (mariusbuescher) [inline grunt/postcss.js]: This is new. `postcss-inline-svg` and `postcss-svgo` are not yet part of the standard set up. Please also remove those plugins from the `package.json`

File: grunt/requirejs.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ module.exports = function( grunt, options ) {
                 mainConfigFile: '<%= srcPath %>components/app/main.js',
                 dir: '<%= distPath %>js',
 
-                fileExclusionRegExp: /\.(njs|scss|gitkeep)/,
+                fileExclusionRegExp: /\.(njs|scss|gitkeep|DS_Store)/,
```
Comment (mariusbuescher) [inline grunt/requirejs.js]: Do we really need this?

File: grunt/scsslint.js [hunk 3]
```diff
@@ -1,20 +0,0 @@
...
-        '!<%= srcPath %>components/sass/vendor/**/*.scss'
-    ],
-
-    options: {
-        config: '.scss-lint.yml',
-        reporterOutput: '<%= testResultPath %>scss-lint-report.xml',
-        compact: true,
-        maxBuffer: NaN
-    }
-};
```
Comment (mariusbuescher) [inline grunt/scsslint.js]: Removing `scss-lint` would be a breaking change. We are developing for 2.x here, so no breaking changes. Please revert the removal of `scss-lint`.

File: src/components/sass/vendor/_normalize.scss [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+/* Hidden
+   ========================================================================== */
+
+/**
+ * Add the correct display in IE 10-.
+ */
+
+[hidden] {
+    display: none;
+}
```
Comment (mariusbuescher) [inline src/components/sass/vendor/_normalize.scss:447]: Why did you move those files? Is a leading underscore really neccesary?

File: docs/deprecated_scss-linting/.stylelintrcComments [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+    "selector-class-pattern": "^[a-z\\-_0-9]+$",                // limit feature                                                                -> stylelintscssrc
+
+    "selector-no-qualifying-type": true,                        // limit feature                                                                -> stylelintscssrc
+
+    "shorthand-property-no-redundant-values": true,             // possible error                           ( stylelint-config-recommended )    -> stylelintscssrc
+
+    "string-quotes": "single",                                  // stylistic issue                                                              -> stylelintscssrc
+  }
+}
```
Comment (mariusbuescher) [inline docs/deprecated_scss-linting/.stylelintrcComments]: This file is of ne need, please remove it.

File: docs/deprecated_scss-linting/.stylelintscssrc [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+
+    "scss/at-extend-no-missing-placeholder": true,
+    "scss/at-import-no-partial-leading-underscore": true,
+    "scss/selector-no-redundant-nesting-selector": true,
+    "scss/at-function-pattern": "^[a-z\\-_0-9]+$",
+    "scss/at-mixin-pattern": "^[a-z\\-_0-9]+$",
+    "scss/dollar-variable-pattern": "^[a-z\\-_0-9]+$"
+  }
+}
+
```
Comment (mariusbuescher) [inline docs/deprecated_scss-linting/.stylelintscssrc]: Same here: not needed anymore, please remove

File: grunt/aliases.yaml [hunk 7]
```diff
@@ -2,9 +2,9 @@ development:
     description: 'Development build task'
     tasks:
         - 'clean:dist'
-        - 'sass:development'
-        - 'postcss:autoprefixer'
         - 'lint'
+        - 'sass:development'
+        - 'postcss:cssDevelopment'
```
Comment (mariusbuescher) [inline grunt/aliases.yaml:7]: Maybe we can combine the `sass` and the `postcss` task to one `styles` task and call it combined, so we do not have to call the whole `development` task when we just want to build the css.

**Review by mariusbuescher — CHANGES_REQUESTED** [review]
So I just reviewed the changes you made. Seems to be more than needed and some things really introduce breaking changes.

File: docs/deprecated_scss-linting/.stylelintrcComments (hunk 5 above)
Comment (kaiwollhaupt) [reply docs/deprecated_scss-linting/.stylelintrcComments]: File removed.

File: docs/deprecated_scss-linting/.stylelintscssrc (hunk 6 above)
Comment (kaiwollhaupt) [reply docs/deprecated_scss-linting/.stylelintscssrc]: File removed.

File: grunt/aliases.yaml (hunk 7 above)
Comment (kaiwollhaupt) [reply grunt/aliases.yaml:7]: Let's call this task stylesDevelopment to recognize if it's building the development or the production CSS.

File: grunt/postcss.js (hunk 1 above)
Comment (kaiwollhaupt) [reply grunt/postcss.js]: Plugins removed.

File: grunt/requirejs.js (hunk 2 above)
Comment (kaiwollhaupt) [reply grunt/requirejs.js]: Plugin grunt-contrib-requirejs threw an exception because it took this file into account. Since i can't reproduce this, i undo this change.

File: grunt/scsslint.js (hunk 3 above)
Comment (kaiwollhaupt) [reply grunt/scsslint.js]: By merging branch feature/add-stylelint to feature/watch-task plugins grunt-contrib-csslint and grunt-scss-lint were removed. Since this is a breaking change I took both plugins in again.

File: src/components/sass/vendor/_normalize.scss (hunk 4 above)
Comment (kaiwollhaupt) [reply src/components/sass/vendor/_normalize.scss:447]: This is a naming convention for partials for which no own CSS file should be created.
http://sass-lang.com/documentation/file.SASS_REFERENCE.html#Partials__partials

File: grunt/aliases.yaml (hunk 7 above)
Comment (mariusbuescher) [reply grunt/aliases.yaml:7]: Please add one task for production as well.

File: src/components/sass/vendor/_normalize.scss (hunk 4 above)
Comment (mariusbuescher) [reply src/components/sass/vendor/_normalize.scss:447]: Thats ok with me.

File: grunt/aliases.yaml [hunk 8]
```diff
@@ -27,6 +27,17 @@ prod:
...
+    description: 'Build development CSS files'
+    tasks:
+        - 'sass:development'
+        - 'postcss:cssDevelopment'
+
+stylesDev:
+    description: 'Shortcut for stylesDevelopment task'
+    tasks:
+        - 'stylesDevelopment'
+
```
Comment (mariusbuescher) [inline grunt/aliases.yaml:40]: Please add one task for production as well.

**Review by mariusbuescher — APPROVED** [review]
